### PR TITLE
Filter zero balance AMM pools 

### PR DESCRIPTION
**Pull request by mohammed7s**
A filter to remove AMMs with zero balances that appear in some of the batch auction jsons.

### CODE REVIEW

**Comment by mohammed0x8** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by bh2smith** [comment]
Code looks good. Have you been seeing pools with zero balance being sent along with the instances? Might also be a good idea to cut them off even earlier in the liquidity collector. I would imagine if you are seeing these that they are likely a single type of liquidity (e.g. balancerV2 pools)

File: src/models/order.py [hunk 1]
```diff
@@ -112,6 +112,7 @@ def from_dict(cls, order_id: str, data: OrderSerializedType) -> Order:
             "buy_amount",
             "is_sell_order",
             "allow_partial_fill",
+            "is_liquidity_order",
```
Comment (bh2smith) [inline src/models/order.py:115]: Are you sure this is a required field?

**Review by bh2smith — APPROVED** [review]

File: src/models/order.py (hunk 1 above)
Comment (mohammed0x8) [reply src/models/order.py:115]: No I am not sure. Just did not feel any less required than other fields.

Comment (bh2smith) [reply src/models/order.py:115]: Being required here means that it is always expected to be provided in an orders data from the instance provided. It's probably best to check this with the instances and their specification.

Comment (mohammed0x8) [reply src/models/order.py:115]: After a bit of thinking: I believe it is required. As a solver, if that was not present in an order then I am not sure what to do with the order. A solver treats `"is_liquidity_order": true` different from `"is_liquidity_order": false`

**Comment by bh2smith** [comment]
Just need to fix the linting errors and this can be merged. Thanks for your contribution!

**Comment by bh2smith** [comment]
Hey @mohammed7s - this is pretty old and awaiting a fix from your side... you gonna do it?

**Review by bh2smith — APPROVED** [review]
Looks amazing! Thanks for your contribution.

**Comment by bh2smith** [comment]
Now you need to format the code with black:

```
black ./
```

Please monitor the success of your changes and make the green check mark!

**Comment by bh2smith** [comment]
I fixed it and made some gross mistake and somehow this thing got merged to main without going through this PR. I will try to revert all these changes, but its going to be a mess and I don't really want to. As of now, your changes are on main branch of this project.

**Pull request merged**